config/config.go

```diff
@@ -2,30 +2,43 @@ package config
 
 import (
 	"fmt"
-	"strings"
+	"io/ioutil"
+	"path/filepath"
 	"sync"
 
 	"github.com/kelseyhightower/envconfig"
-	"github.com/mesg-foundation/core/version"
 	"github.com/mesg-foundation/core/x/xstrings"
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
 )
 
 const envPrefix = "mesg"
 
 var (
 	instance *Config
 	once     sync.Once
+
+	logformats = []string{"text", "json"}
 )
 
-// Path to a dedicated directory for Core
-// TODO: Path should be reverted to a const when the package database is renovated
-var Path = "/mesg"
+type plugin struct {
+	Name string
+	Path string
+	ID   string
+}
 
 // Config contains all the configuration needed.
 type Config struct {
 	Server struct {
 		Address string
+
+		Plugins []plugin
```

Comment:
I'm still not convinced using a plugin with a path. Also I was thinking more something like that:
```
Plugin struct {
  Resolver string
}
```

Like that we can identify each plugins

Reply:
I did ID and Path in plugin struct because of new users - they don't want to "hack" with id. Even for me, it would be easier to develop - just giving the path and edit code and I don't need to think about what I have changed in each plugin or core etc...

> Like that we can identify each plugins

yes, but we can easily do it in the code and by giving slice of plugins with name it gives us more flexibility for next plugins (we don't need to change the config). Moreover, we don't limit plugins that someone may start with the core.

In the feature our plugins may start with special prefix or we can even come up with diffrent apporach and resolver is good name so I don't want to take it from users ([Don't steal good names from the user.](https://blog.golang.org/package-names)). 

Even we were not sure about the name (resolver vs discover).